src/core/FileManagerServices.cpp

```diff
@@ -58,24 +58,29 @@ void FileManagerServices::reveal(const QFileInfo item)
 
 bool FileManagerServices::supportsSelectOption(const QString& fileManager)
 {
+	if (selectOptionCache.contains(fileManager)) { return selectOptionCache[fileManager]; }
 #if !defined(_WIN32) && !defined(__APPLE__)
 
 	QProcess process;
 	process.start(fileManager, {"--help"});
 	process.waitForFinished(3000);
 
 	QString output = process.readAllStandardOutput() + process.readAllStandardError();
-	return output.contains("--select");
+	bool supports = output.contains("--select");
+
+	selectOptionCache.insert(fileManager, supports);
+	return supports;
 #else
+	selectOptionCache.insert(fileManager, true);
 	return true;
 #endif
 }
 
 QString FileManagerServices::getDefaultFileManager()
 {
-	if (defaultFileManager.has_value()) { return defaultFileManager.value(); }
+	if (fileManagerCache.has_value()) { return fileManagerCache.value(); }
 #if defined(_WIN32)
-	defaultFileManager = "explorer";
+	fileManagerCache = "explorer";
 #elif defined(__APPLE__)
 	defaultFileManager = "open";
```

Comment:
I believe you forgot to rename `defaultFileManager` here and in Linux section when you refactored this.